transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -1118,6 +1142,20 @@ private void callHandlerCallbackLater(AbstractChannelHandlerContext ctx, boolean
         }
     }
 
+    private void wakeUpForAutoFlush() {
+        if (autoFlush && ENQUEUE_WAKEUP_TASK_UPDATER.compareAndSet(this, 0, 1)) {
+            if (null == wakeupTask) {
+                wakeupTask = new Runnable() {
+                    @Override
+                    public void run() {
+                        ENQUEUE_WAKEUP_TASK_UPDATER.set(DefaultChannelPipeline.this, 0);
+                    }
+                };
+            }
+            channel().eventLoop().execute(wakeupTask);
+        }
+    }
+
```

Comment:
Working on it. Running into some exceptions when trying `AUTO_FLUSH` that I'll need to sort out:

```
[nioEventLoopGroup-6-2] WARN io.netty.channel.SingleThreadEventLoop - A task executed after event loop iteration raised an exception.
java.lang.IllegalStateException: Queue full
	at java.util.AbstractQueue.add(AbstractQueue.java:98)
	at io.netty.channel.SingleThreadEventLoop.onEventLoopIteration(SingleThreadEventLoop.java:95)
	at io.netty.channel.AbstractChannel$AutoFlushTask.run(AbstractChannel.java:1094)
	at io.netty.channel.SingleThreadEventLoop.afterRunningAllTasks(SingleThreadEventLoop.java:170)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:348)
	at io.netty.channel.SingleThreadEventLoop.runAllTasks(SingleThreadEventLoop.java:155)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:393)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:771)
	at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:145)
	at java.lang.Thread.run(Thread.java:745)
```